### PR TITLE
Minor hygenic changes related to Executors

### PR DESCRIPTION
**Pull request by carl-mastrangelo**

### CODE REVIEW

**Comment by carl-mastrangelo** [comment]
ping

File: auth/src/main/java/io/grpc/auth/ClientAuthInterceptor.java [hunk 1]
```diff
@@ -64,12 +64,11 @@
...
   private Map<String, List<String>> lastMetadata;
-  // TODO(louiscryan): refresh token asynchronously with this executor.
-  private Executor executor;
 
-  public ClientAuthInterceptor(Credentials credentials, Executor executor) {
+  public ClientAuthInterceptor(
+      Credentials credentials, @SuppressWarnings("unused") Executor executor) {
     this.credentials = Preconditions.checkNotNull(credentials);
-    this.executor = Preconditions.checkNotNull(executor);
+    // TODO(louiscryan): refresh token asynchronously with this executor.
```
Comment (zhangkun83) [inline auth/src/main/java/io/grpc/auth/ClientAuthInterceptor.java]: When we fix the TODO, wouldn't we go back to saving the executor? I don't see it's necessary for now to make this change.

Comment (carl-mastrangelo) [inline auth/src/main/java/io/grpc/auth/ClientAuthInterceptor.java]: Yes, it will be added back in when its used.

**Comment by zhangkun83** [comment]
LGTM

**Pull request merged**